src/Tasks/Microsoft.NET.Build.Tasks/GetDependsOnNETStandard.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Build.Framework;
+using System;
+using System.IO;
+using System.Security;
+
+namespace Microsoft.NET.Build.Tasks
+{
+    /// <summary>
+    /// Determines the assembly version to use for a given semantic version.
+    /// </summary>
+    public partial class GetDependsOnNETStandard : TaskBase
+    {
+        private const string NetStandardAssemblyName = "netstandard";
+        /// <summary>
+        /// Set of reference items to analyze.
+        /// </summary>
+        [Required]
+        public ITaskItem[] References { get; set; }
+
+        /// <summary>
+        /// True if any of the references depend on netstandard.dll
+        /// </summary>
+        [Output]
+        public bool DependsOnNETStandard { get; set; }
+
+        protected override void ExecuteCore()
+        {
+            foreach (var reference in References)
+            {
+                var referenceSourcePath = ItemUtilities.GetSourcePath(reference);
+
+                if (referenceSourcePath != null && File.Exists(referenceSourcePath))
+                {
+                    try
+                    {
+                        if (DependsOnNETStandard = GetFileDependsOnNETStandard(referenceSourcePath))
```

Comment:
Similar nit here. Factor out a method that loops over references and returns true/false and then assign result of that to DependsOnNETStandard.